kafka-connect-s3/src/test/java/io/confluent/connect/s3/S3ProxyTest.java

```diff
@@ -54,7 +72,18 @@
   @Before
   public void setUp() throws Exception {
     super.setUp();
-    storage = new S3Storage(connectorConfig, url, S3_TEST_BUCKET_NAME, null);
+    storage = new S3Storage(connectorConfig, url);
+    AmazonS3 s3 = Whitebox.getInternalState(storage, "s3");
```

Comment:
That isn't really a mocking internal state, but rather getting a private fields' states to assert their behavior.
In this case, I want to read the retry policy which has been used to configure and build a client. Since we agreed to make `io.confluent.connect.s3.storage.S3Storage#newFullJitterRetryPolicy` method protected, and extending the `S3Storage` class in this test is not the most elegant way I use `Whitebox` to read private fields here.

Reply:
That's what I meant by mocking. In general, including capturing and inspecting internal state too. 

I suggested `protected` because this also gives package private access. So if the test is in the right package (here `io.confluent.connect.s3.storage`) then you have access without inheriting the class. 

This package doesn't exist in tests and you'll have to create it if you go down that path. If that's all, it still sounds better than whitebox.

Reply:
Yes, I like this suggestion - same-package access level will make it much cleaner. Also, after moving `S3ProxyTest` to `io.confluent.connect.s3.storage` all three methods(`newClientConfiguration`, `newFullJitterRetryPolicy`, `newS3Client`) can be protected.

Reply:
Great! Let's not change access scope to existing methods. Only the new one. Feel free to add all the needed unit tests in S3StorageTest (I assume) regardless of scope (public, or protected).